.taskcluster.yml

```diff
@@ -1,125 +1,264 @@
+                      then: '${event.pull_request.base.repo.html_url}'
+                      else:
+                          $if: 'tasks_for in ["cron", "action"]'
+                          then: '${repository.url}'
+              repoUrl:
+                  $if: 'tasks_for in ["github-push", "github-release"]'
+                  then: '${event.repository.html_url}'
+                  else:
+                      $if: 'tasks_for == "github-pull-request"'
+                      then: '${event.pull_request.head.repo.html_url}'
+                      else:
+                          $if: 'tasks_for in ["cron", "action"]'
+                          then: '${repository.url}'
+              project:
+                  $if: 'tasks_for in ["github-push", "github-release"]'
+                  then: '${event.repository.name}'
+                  else:
+                      $if: 'tasks_for == "github-pull-request"'
+                      then: '${event.pull_request.head.repo.name}'
+                      else:
+                          $if: 'tasks_for in ["cron", "action"]'
+                          then: '${repository.project}'
+              head_branch:
+                  $if: 'tasks_for == "github-pull-request"'
+                  then: ${event.pull_request.head.ref}
+                  else:
+                      $if: 'tasks_for == "github-push"'
+                      then: ${event.ref}
+                      else:
+                          $if: 'tasks_for == "github-release"'
+                          then: '${event.release.target_commitish}'
+                          else:
+                              $if: 'tasks_for in ["cron", "action"]'
+                              then: '${push.branch}'
+              head_sha:
+                  $if: 'tasks_for == "github-push"'
+                  then: '${event.after}'
+                  else:
+                      $if: 'tasks_for == "github-pull-request"'
+                      then: '${event.pull_request.head.sha}'
+                      else:
+                          $if: 'tasks_for == "github-release"'
+                          then: '${event.release.tag_name}'
+                          else:
+                              $if: 'tasks_for in ["cron", "action"]'
+                              then: '${push.revision}'
+              ownTaskId:
+                  $if: '"github" in tasks_for'
+                  then: {$eval: as_slugid("decision_task")}
+                  else:
+                      $if: 'tasks_for == "cron"'
+                      then: '${ownTaskId}'
+          in:
+              $if: 'tasks_for in ["github-pull-request", "github-release", "action", "cron"] || tasks_for == "github-push" && head_branch[:10] != "refs/tags/"'
+              then:
+                  $let:
+                      level:
+                          $if: 'tasks_for in ["github-push", "github-release", "action", "cron"] && repoUrl == "https://github.com/mozilla-mobile/firefox-tv"'
```

Comment:
Perhaps nothing to change here but it kind of sucks how tied to GitHub all these configuration files are. :\ I wonder if we'd want to move our projects back to m-c at some point.

Reply:
This file is largely identical between all the taskgraph using github project, and the differences are mostly (only mostly, unfortunately) confined to the stanza at the beginning of this file. We have a similar somewhat standardized config for hg.mozilla.org products, that more or less shares the same initial stanza. So, migrate to hg.mozilla.org shouldn't be too painful in that regard.

---
A couple of additional details:
- We are just starting to use taskgraph outside of mozilla-central, so are still figuring out what abstraction we need for specifying `.taskcluster.yml` files, the variation beyond the initial stanza should go down over time.
- We've talked about various ways to avoid needing a mostly duplicated `.taskcluster.yml`, and being able to just have a simpler format for specifying things. We have some ideas on how to specify that, but require support from the taskcluster platform. Given that that team is dealing with several migration at the moment, it will be awhile before something better is implemented.

Reply:
Yeah, it sounds like the previous intent of TaskCluster was to act as the CI system (that just happened to talk to Github as a backend), but teams were asking for access to Github properties/features faster than the TC team planned, so this iteration of TaskCluster has more of a "pass-through" role